### DISCUSSION TITLE
Storage size limitations?

### DISCUSSION THREAD

**Discussion by SKempin** [react-native-async-storage/async-storage#780] [General]
### Proposal

I'm looking at file storage options using Expo and it seems that Async Storage is the only supported option. However it is documented that there is a [6MB limit on Android](https://react-native-async-storage.github.io/async-storage/docs/limits#android) for this. I'm curious, how do native apps like Netflix/Spotify (for example) utilise hundreds of MB's of storage space? Is this not possible with Expo / React Native? I need to store around 50mb of JSON data.

Thanks

**Comment by krizzu** [comment]
The 6MB comes from very old days, I don't know why exactly the 6MB limit was put in the first place. SQLite handles much much more than that. You could have increased [by setting a property with desired storage size](https://react-native-async-storage.github.io/async-storage/docs/advanced/db_size). However, I'm not familiar with Expo so I can't advise on how to set it there.

**Comment by juanpb1** [comment]
https://react-native-async-storage.github.io/async-storage/docs/advanced/db_size
